app/models/shipit/deploy_spec/npm_discovery.rb

```diff
@@ -79,7 +92,67 @@ def package_version
         JSON.parse(package_json.read)['version']
       end
 
+      def publish_config
+        JSON.parse(package_json.read)['publishConfig']
+      end
+
+      def publish_config_access
+        publish_config['access'] || PRIVATE
+      end
+
+      def package_name
+        JSON.parse(package_json.read)['name']
+      end
+
+      def scoped_package?
+        package_name.start_with?('@shopify')
+      end
+
+      def publish?
+        return false if publish_config.blank?
+        return false unless valid_publish_config_access?
+        return false unless package_scoped_when_private?
+      end
+
+      def valid_publish_config_access?
+        return false unless VALID_ACCESS.include?(publish_config_access)
+        true
+      end
+
+      def package_scoped_when_private?
+        private_package = publish_config_access == PRIVATE
+        return false if private_package && !scoped_package?
+        true
+      end
+
+      def local_npmrc
+        file(".npmrc")
+      end
+
+      def npmrc_contents(registry)
+        "always-auth=true\n#{registry}"
+      end
+
+      def registry
+        if publish_config_access == "public"
+          return scoped_package? ? "@shopify:registry=#{NPM_REGISTRY}" : "registry=#{NPM_REGISTRY}"
+        end
+
+        PACKAGE_CLOUD_REGISTRY
+      end
+
+      def generate_local_npmrc
+        File.delete(local_npmrc) if File.exist?(local_npmrc)
```

Comment:
Not sure if this is detrimental or not. The reason i'm doing this to ensure a cached npmrc from a previous deploy isn't used. We can hit the `git clear cache`, but if we forget to do this and a private deploy goes public we will be 😞

Reply:
>  We can either leave it be, or try appending the correct scope info to it.

I don't know of a case where a project would required having a `.npmrc`. If it does the project is doing something very specific. We should defer the contents of the `.npmrc` to the project if it has an `.npmrc`